docs/resources/samples/filter-table-get-visible-range.md

```diff
@@ -0,0 +1,127 @@
+* Assemble an object with following JSON structure and return for a future action to use.
+
+### Sample JSON
+
+Each key represents a unique value of a table. Each array instance represents the row that is visible when the corresponding filter is applied.
+
+```json
+{
+  "Station-1": [{
+    "Station": "Station-1",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Debra Berger",
+    "Reason": ""
+  }, {
+    "Station": "Station-1",
+    "Shift": "Afternoon",
+    "Date": "27-Oct-20",
+    "Responsible": "Delia Dennis",
+    "Reason": ""
+  }, {
+    "Station": "Station-1",
+    "Shift": "Afternoon",
+    "Date": "28-Oct-20",
+    "Responsible": "Lidia Holloway",
+    "Reason": ""
+  }],
+  "Station-2": [{
+    "Station": "Station-2",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Gerhart Moller",
+    "Reason": ""
+  }, {
+    "Station": "Station-2",
+    "Shift": "Afternoon",
+    "Date": "28-Oct-20",
+    "Responsible": "Grady Archie",
+    "Reason": ""
+  }],
+  "Station-3": [{
+    "Station": "Station-3",
+    "Shift": "Morning",
+    "Date": "27-Oct-20",
+    "Responsible": "Isaiah Langer",
+    "Reason": ""
+  }]
+}
+```
+
+## Solution
+
+The solution has a single Office Script:
+
+* Performs transformation as described previously and extracts required data
+
+## Office Script
+
+The following script filters a table and gets the visible range.
+
+Download the sample file <a href="table-filter.xlsx">table-filter.xlsx</a> and use it with this script to try it out yourself!
+
+```ts
+function main(workbook: ExcelScript.Workbook): ReturnTemplate {
+  const table1 = workbook.getTable("Table1");
+  const keyColumnValues: string [] = table1.getColumnByName('Station').getRangeBetweenHeaderAndTotal().getValues().map(v => v[0] as string);
+  // const uniqueKeys= [...Array.from(new Set(keyColumnValues))];
+  const uniqueKeys = keyColumnValues.filter((v, i, a) => a.indexOf(v) === i);
+
+  console.log(uniqueKeys);
+  const returnObj: ReturnTemplate = {}
+
+  uniqueKeys.forEach((key: string) => {
+    table1.getColumnByName('Station').getFilter()
+      .applyValuesFilter([key]);
+    const rangeView = table1.getRange().getVisibleView();
+    returnObj[key] = returnObjectFromValues(rangeView.getValues() as string[][]);
+  })
+  table1.getColumnByName('Station').getFilter().clear();
+  console.log(JSON.stringify(returnObj));
+  return returnObj
+}
+
+function returnObjectFromValues(values: string[][]): BasicObj[] {
+  let objArray = [];
+  let objKeys: string[] = [];
+  for (let i=0; i < values.length; i++) {
+    if (i===0) {
+      objKeys = values[i]
+      continue;
+    }
+    let obj = {}
+    for (let j=0; j < values[i].length; j++) {
+      obj[objKeys[j]] = values[i][j]
+    }
+    objArray.push(obj);
+  }
+  return objArray;
+}
+
+interface BasicObj {
+  [key: string] : string
+}
+
+interface ReturnTemplate {
+  [key: string]: BasicObj[]
+}
+```
```

Comment:
**Edit:** Wrote this before my comment about making this a generic Map sample.
Rather than having generic names, I think using terms specific to the scenario would help increase comprehension. Maybe StationList and Station?

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)